### PR TITLE
Exchanging messages with Gnokey Mobile

### PR DESCRIPTION
**Pull request by iuricmp**
- Prepares dSocial to send and to receive Linking Parameters from the operational system.
- Updates "@gnolang/gnonative" to "3.0.0"

How to test:
1. Open dSocial and Gnokey mobile
2. Create accounts in both of them
3. Try to Post a message using dSocial
4. After pressing the Post button, dSocial will open Gnokey Mobile and expect you select a key to proceed.
5. After selecting a key in Gnokey, it will navigate back do dSocial.
6. dSocial will receive the key address you selected and will use it to create a unsigned transaction.
7. The unsigned transaction will be sent to Gnokey to be signed.
8. Gnokey requires you to select a key to sign the transaction. Use the same key you selected before.
9. The signed transaction is sent back to dSocial.
10. dSocial will broadcast the transaction to be inserted in the blockchain.

### CODE REVIEW

File: mobile/redux/features/linkingSlice.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+    setTimeout(() => {
+        const params = [`tx=${encodeURIComponent(res.txJson)}`, `address=${callerAddressBech32}`, `client_name=dSocial`, `reason=Post a message`];
+        Linking.openURL('land.gno.gnokey://tosign?' + params.join('&'))
+    }, 500)
+
+    return res;
+})
+
+type MakeTxParams = {
```
Comment (jefft0) [inline mobile/redux/features/linkingSlice.ts:54]: I suggest to rename this to `MakeCallTxParams` . Each transaction type has its own params. For example, [send](https://github.com/gnolang/gnonative/blob/4bb275d89768062b2281613d9ed9fd925ad00e88/expo/src/api/types.ts#L74-L81) is a different transaction type and doesn't have params like `fnc`.

**Comment by jefft0** [comment]
To fix the Following bug `user.address.toString()` should be `user.bech32`, on this line:
https://github.com/gnolang/dsocial/blob/inter-app-communication/mobile/components/view/account/index.tsx#L64

File: mobile/redux/features/accountSlice.ts [hunk 2]
```diff
@@ -13,46 +13,50 @@ const initialState: CounterState = {
...
-    }
+  const gnonative = thunkAPI.extra.gnonative;
 
-    userCache.invalidateCache();
-  } catch (error) {
-    console.error("on saving avatar", error);
-  }
+  const gasFee = "1000000ugnot";
+  const gasWanted = BigInt(10000000);
+  const args: Array<string> = ["Av  atar", String(`data:${mimeType};base64,` + base64)];
```
Comment (jefft0) [inline mobile/redux/features/accountSlice.ts:57]:
```suggestion
  const args: Array<string> = ["Avatar", String(`data:${mimeType};base64,` + base64)];
```

**Review by jefft0 — APPROVED** [review]
Tested on Android and iOS simulators. See my suggested fix for "Avatar".

**Comment by iuricmp** [comment]
That’s it.

**Pull request merged**